tests/pytests/test_tags.py

```diff
@@ -374,3 +374,271 @@ def test_empty_suffix_withsuffixtrie(env):
+                  'NOCONTENT', 'DIALECT', 5)
+    env.assertEqual(res, [1, '{doc}:3'])
+
+    # Test prefix
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{a-b-c*}', 'DIALECT', 5)
+    env.assertEqual(res, "TAG:@tag {\n  PREFIX{a-b-c*}\n}\n")
+
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{a-b-c\\*}', 'DIALECT', 5)
+    env.assertEqual(res, 'TAG:@tag {\n  a-b-c*\n}\n')
+
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{abc\\*yxv*}', 'DIALECT', 5)
+    env.assertEqual(res, 'TAG:@tag {\n  PREFIX{abc*yxv*}\n}\n')
+
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{abc:?\\*yxv*}=>{$weight:3.4}',
+                  'PARAMS', '2', 'abc', 'hello', 'DIALECT', 5)
+    env.assertEqual(res, 'TAG:@tag {\n  PREFIX{abc:?*yxv*}\n} => { $weight: 3.4; }\n')
+
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{abc:?*}=>{$weight:3.4}',
+                  'PARAMS', '2', 'abc', 'hello', 'DIALECT', 5)
+    env.assertEqual(res, 'TAG:@tag {\n  PREFIX{abc:?*}\n} => { $weight: 3.4; }\n')
+
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{$abc*}=>{$weight:3.4}',
+                  'PARAMS', '2', 'abc', 'hello', 'DIALECT', 5)
+    env.assertEqual(res, 'TAG:@tag {\n  PREFIX{hello*}\n} => { $weight: 3.4; }\n')
+
+    res = env.cmd('FT.SEARCH', 'idx', '@tag:{abc:*}=>{$weight:3.4}',
+                  'NOCONTENT', 'SORTBY', 'id', 'ASC', 'DIALECT', 5)
+    env.assertEqual(res, [4, '{doc}:1', '{doc}:3', '{doc}:4', '{doc}:7'])
+
+
+    # Test suffix
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{*a-b-c}', 'DIALECT', 5)
+    env.assertEqual(res, "TAG:@tag {\n  SUFFIX{*a-b-c}\n}\n")
+
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{\\*a-b-c}', 'DIALECT', 5)
+    env.assertEqual(res, 'TAG:@tag {\n  *a-b-c\n}\n')
+
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{*abc\\*yxv}', 'DIALECT', 5)
+    env.assertEqual(res, 'TAG:@tag {\n  SUFFIX{*abc*yxv}\n}\n')
+
+    res = env.cmd('FT.SEARCH', 'idx', '@tag:{*xyz:2}=>{$weight:3.4}',
+                  'NOCONTENT', 'SORTBY', 'id', 'ASC', 'DIALECT', 5)
+    env.assertEqual(res, [4, '{doc}:2', '{doc}:3', '{doc}:4', '{doc}:7'])
+
+    res = env.cmd('FT.SEARCH', 'idx', '@tag:{*$param}',
+                  'PARAMS', '2', 'param', 'xyz:2',
+                  'NOCONTENT', 'SORTBY', 'id', 'ASC', 'DIALECT', 5)
+    env.assertEqual(res, [4, '{doc}:2', '{doc}:3', '{doc}:4', '{doc}:7'])
+
+    # Test infix
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{*a-b-c*}', 'DIALECT', 5)
+    env.assertEqual(res, "TAG:@tag {\n  INFIX{*a-b-c*}\n}\n")
+
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{\\*a-b-c\\*}', 'DIALECT', 5)
+    env.assertEqual(res, 'TAG:@tag {\n  *a-b-c*\n}\n')
+
+    res = env.cmd('FT.EXPLAIN', 'idx', '@tag:{*abc\\*yxv:*}', 'DIALECT', 5)
+    env.assertEqual(res, 'TAG:@tag {\n  INFIX{*abc*yxv:*}\n}\n')
+
+    res = env.cmd('FT.SEARCH', 'idx', '@tag:{*@mail.*}=>{$weight:3.4}',
+                  'NOCONTENT', 'DIALECT', 5)
+    env.assertEqual(res, [1, '{doc}:5'])
+
+    res = env.cmd('FT.SEARCH', 'idx', '@tag:{*$param*}=>{$weight:3.4}',
```

Comment:
What about escaping a `$` so it is a literal and not a param?

Reply:
Tests included in [e079ac4](https://github.com/RediSearch/RediSearch/pull/4433/commits/e079ac449494870351a6d77f3e0430de83d359e7) and [3195bcc](https://github.com/RediSearch/RediSearch/pull/4433/commits/3195bcc6ddc17b86b5dbc3a694eaa64da5b3482c)